strawberryfields/backends/fockbackend/tests/test_fock_probabilities.py

```diff
@@ -0,0 +1,99 @@
+    @pytest.mark.parametrize("phase_alpha", PHASE_ALPHAS)
+    def test_prob_fock_of_coherent_state(self, setup_backend, mag_alpha, phase_alpha, cutoff, tol):
+        """Tests that probabilities of particular Fock states |n> are correct for a coherent state."""
+
+        backend = setup_backend(1)
+
+        alpha = mag_alpha * np.exp(1j * phase_alpha)
+        ref_state = np.array([np.exp(-0.5 * np.abs(alpha) ** 2) * alpha ** n / np.sqrt(factorial(n)) for n in range(cutoff)])
+        ref_probs = np.abs(ref_state) ** 2
+        backend.prepare_coherent_state(alpha, 0)
+        state = backend.state()
+        prob_n = [state.fock_prob([n]) for n in range(cutoff)]
+        assert np.allclose(prob_n, ref_probs, atol=tol, rtol=0)
+
+class TestFockRepresentation:
+    """Tests that make use of the Fock basis representation."""
+
+    @pytest.mark.parametrize("mag_alpha", MAG_ALPHAS)
+    @pytest.mark.parametrize("phase_alpha", PHASE_ALPHAS)
+    def test_all_fock_probs_of_coherent_state(self, setup_backend, mag_alpha, phase_alpha, cutoff, tol):
+        """Tests that the numeric probabilities in the full Fock basis are correct for a coherent state."""
+
+        backend = setup_backend(1)
+
+        alpha = mag_alpha * np.exp(1j * phase_alpha)
+        ref_state = np.array([np.exp(-0.5 * np.abs(alpha) ** 2) * alpha ** n / np.sqrt(factorial(n)) for n in range(cutoff)])
+        ref_probs = np.abs(ref_state) ** 2
+        backend.prepare_coherent_state(alpha, 0)
+        state = backend.state()
+        probs = state.all_fock_probs().flatten()
+        assert np.allclose(probs, ref_probs, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("mag_alpha", MAG_ALPHAS)
+    @pytest.mark.parametrize("phase_alpha", PHASE_ALPHAS)
+    def test_prob_fock_of_nongaussian_state(self, setup_backend, mag_alpha, phase_alpha, cutoff, tol):
+        """Tests that Fock probabilities are correct for a specific family of nongaussian states
+        (tensor product of coherent state and fock state)."""
+
+        backend = setup_backend(2)
+
+        alpha = mag_alpha * np.exp(1j * phase_alpha)
+        ref_state = np.array([np.exp(-0.5 * np.abs(alpha) ** 2) * alpha ** n / np.sqrt(factorial(n)) for n in range(cutoff)])
+        ref_probs = np.abs(ref_state) ** 2
+        backend.prepare_coherent_state(alpha, 0)
+        backend.prepare_fock_state(cutoff // 2, 1)
+        state = backend.state()
+        prob_n = [state.fock_prob([n, cutoff // 2]) for n in range(cutoff)]
+        assert np.allclose(prob_n, ref_probs, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("mag_alpha", MAG_ALPHAS)
+    @pytest.mark.parametrize("phase_alpha", PHASE_ALPHAS)
+    def test_all_fock_state_probs_of_product_coherent_state(self, setup_backend, mag_alpha, phase_alpha, cutoff, tol):
+        """Tests that the numeric probabilities in the full Fock basis are correct for a two-mode product coherent state."""
+
+        if mag_alpha == 0.:
+          pass
```

Comment:
Instead of using `pass`, better to skip the test with a useful message via
```python
pytest.skip("no displacement, test skipped")
```

Reply:
Not sure why this was skipped. There's no harm checking that it gets the right answer for the zero displacement case.
I've just removed the conditional branch.

Reply:
oh, strange. Must have been a copy/paste during the making of the old test suite?